public/static/js/legacyMain.js

```diff
@@ -80,14 +80,40 @@ const listenScrollToTopButton = () => {
   });
 };
 
-const detectEnviromentAndSetDownloadOptions = () => {
+const detectEnviromentAndSetDownloadOptions = async () => {
+  /**
+   * This function is used to get the CPU bitness of the user's device
+   * on Windows 11. For more information, see:
+   *
+   * @see
+   * [MDN](https://developer.mozilla.org/en-US/docs/Web/API/NavigatorUAData/getHighEntropyValues)
+   * and
+   * [MSFT](https://learn.microsoft.com/en-us/microsoft-edge/web-platform/how-to-detect-win11)
+   *
+   * @returns
+   * 32 or 64, depending on the CPU bitness of the user's device.
+   * If not supported, it returns undefined.
+   */
+  async function getCPUBitnessForWin11() {
+    if (
+      navigator.userAgentData &&
+      navigator.userAgentData.getHighEntropyValues
+    ) {
+      return await navigator.userAgentData.getHighEntropyValues(['bitness']);
+    }
+    return undefined;
+  }
+
+  const getWin11CPUProcessorBitness = getCPUBitnessForWin11();
```

Comment:
I don't think Win11 is relevant here. This works for any OS and any browser that supports the `getHighEntropyValues()` API 

I tested on Edge and Chrome on Windows and macOS to confirm.

Reply:
> I don't think Win11 is relevant here. This works for any OS and any browser that supports the `getHighEntropyValues()` API

Any browser? Are you sure? From MDN document, it seems FF won't be supported.
![image](https://user-images.githubusercontent.com/52018749/225185450-3b0efc92-fa02-4597-8611-0b4e8cebf885.png)

What's more: Maybe we still consider IE6 - IE11? They won't support such functions.....

Reply:
I think you misunderstood. I said "any browser that supports".

More importantly, it works on every OS – not just Win11.

Reply:
Yeah. To be honest, I wish you could just adopt my snippet, @MaledongGit it would make things easier. I know you'd probably want to keep the code in the way you done, but I'm sadly not going to approve it this way. If we're touching this, I believe we should keep it tidy and well documented.